elixir/native/extism_nif/src/lib.rs

```diff
@@ -0,0 +1,51 @@
+use rustler::{Error};
+use extism::{Plugin};
+use std::str;
+use serde_json;
+
+mod atoms {
+    rustler::atoms! {
+        ok,
+        error,
+        eof,
+        unknown // Other error
+    }
+}
+
+// fn load(env: Env, _: Term) -> bool {
+//     true
+// }
+
+ #[rustler::nif]
+fn plugin_new_with_manifest(manifest_payload: String) -> Result<isize, Error> {
+    match serde_json::from_str(&manifest_payload) {
+        Ok(manifest) => {
+            match Plugin::new_with_manifest(&manifest, false) {
+                Err(_e) => Err(Error::Term(Box::new("Could not load plugin"))),
+                Ok(plugin) => Ok(plugin.as_isize())
+            }
+        },
+        Err(_e) => Err(Error::Term(Box::new("Could not parse manifest")))
+    }
+}
+
+ #[rustler::nif]
+fn call_plugin(plugin_iz: isize, name: String, input: String) -> Result<String, Error> {
```

Comment:
Thanks for working on this - it looks great!

My one early bit of feedback is the other SDKs don't require input/output to be UTF8 compatible (`Vec<u8>` instead of `String` here) - I'm not opposed to making an exception if you think this is a nicer API for elixir or if there's some other reason this works better.

Reply:
Let me see how ergonomic it is on the erlang side. We have a wrapper around the `Native` module that it exposes so we can always transform types coming in and out if we need.